app/gui/src/project-view/components/TopBar.vue

```diff
@@ -1,36 +1,36 @@
 <script setup lang="ts">
+import ActionButton from '@/components/ActionButton.vue'
+import ControlGroup from '@/components/ControlGroup.vue'
 import ExtendedMenu from '@/components/ExtendedMenu.vue'
 import NavBreadcrumbs from '@/components/NavBreadcrumbs.vue'
-import RecordControl from '@/components/RecordControl.vue'
 import SelectionMenu from '@/components/SelectionMenu.vue'
-import UndoRedoButtons from './UndoRedoButtons.vue'
+import ZoomControl from '@/components/ZoomControl.vue'
+import { injectGraphSelection } from '@/providers/graphSelection'
 
-const showCodeEditor = defineModel<boolean>('showCodeEditor', { required: true })
-const showDocumentationEditor = defineModel<boolean>('showDocumentationEditor', { required: true })
-const props = defineProps<{
-  zoomLevel: number
-}>()
-const emit = defineEmits<{
-  fitToAllClicked: []
-  zoomIn: []
-  zoomOut: []
-}>()
+const projectNameEdited = defineModel<boolean>('projectNameEdited', { default: false })
+const props = defineProps<{ zoomLevel: number }>()
+const selection = injectGraphSelection()
 </script>
 
 <template>
   <div class="TopBar">
-    <NavBreadcrumbs />
-    <RecordControl />
-    <UndoRedoButtons />
-    <SelectionMenu />
-    <ExtendedMenu
-      v-model:showCodeEditor="showCodeEditor"
-      v-model:showDocumentationEditor="showDocumentationEditor"
-      :zoomLevel="props.zoomLevel"
-      @fitToAllClicked="emit('fitToAllClicked')"
-      @zoomIn="emit('zoomIn')"
-      @zoomOut="emit('zoomOut')"
-    />
+    <ExtendedMenu />
+    <NavBreadcrumbs v-model:projectNameEdited="projectNameEdited" />
+    <ControlGroup>
+      <ActionButton class="redButton" action="graph.refreshExecution" />
+      <ActionButton class="redButton" action="graph.recomputeAll" />
+    </ControlGroup>
+    <ControlGroup>
+      <ActionButton action="graph.undo" />
+      <ActionButton action="graph.redo" />
+    </ControlGroup>
+    <SelectionMenu v-if="selection.selected.size > 1" />
```

Comment:
SelectionMenu's root is `div`, but might be `ControlGroup` perhaps?

Reply:
Right now `ControlGroup` wraps each child into its own element. It's just meant to hold buttons and style them properly. I can see that working eventually, but I think that's a bit too much refactoring work to do for now.